wikipedia-sentences/app.py

```diff
@@ -4,67 +4,85 @@
 import os
 import sys
 
+import copy
 import click
-from jina.flow import Flow
+import shutil
+import numpy as np
+from jina import Flow, Document, DocumentArray
 from jina.logging.profile import TimeContext
 
 
-MAX_DOCS = int(os.environ.get('JINA_MAX_DOCS', 50))
+from transformer import MyTransformer
+from indexer import NumpyIndexer
 
+MAX_DOCS = int(os.environ.get('JINA_MAX_DOCS', 50))
 
 def config():
-    os.environ['JINA_DATA_FILE'] = os.environ.get('JINA_DATA_FILE', 'data/toy-input.txt')
+    os.environ['JINA_DATA_FILE'] = os.environ.get('JINA_DATA_FILE', 'data/input.txt')
     os.environ['JINA_WORKSPACE'] = os.environ.get('JINA_WORKSPACE', 'workspace')
     os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45678))
 
 
 def print_topk(resp, sentence):
-    for d in resp.search.docs:
-        print(f'Ta-Dah🔮, here are what we found for: {sentence}')
-        for idx, match in enumerate(d.matches):
+    for doc in resp.data.docs:
+        print(f"Ta-Dah🔮, here's what we found for: {sentence}")
+        for idx, match in enumerate(doc.matches):
 
             score = match.score.value
             if score < 0.0:
                 continue
             print(f'> {idx:>2d}({score:.2f}). {match.text}')
 
-
 def index(num_docs):
-    f = Flow().load_config('flows/index.yml')
-
-    with f:
-        data_path = os.path.join(os.path.dirname(__file__), os.environ.get('JINA_DATA_FILE', None))
-        num_docs = min(num_docs, len(open(data_path).readlines()))
+    flow = Flow().add(uses=MyTransformer).add(uses=NumpyIndexer)
+    data_path = os.path.join(os.path.dirname(__file__), os.environ.get('JINA_DATA_FILE', None))
+    with flow, open(data_path) as fp:
+        docs = DocumentArray.from_ndarray(np.array(fp.readlines()))
```

Comment:
This is probably super dumb but where is that `from_ndarray` coming? is not part of the `DocumentArray` right?

Reply:
it is part of DocumentArray API, see https://github.com/jina-ai/jina/blob/master/.github/2.0/cookbooks/Document.md#construct-from-json-csv-ndarray-and-files

Reply:
Thanks!